### DISCUSSION TITLE
Images extracted are mirrored on x-axis.

### DISCUSSION THREAD

**Discussion by yashpundir** [jsvine/pdfplumber#1074] [Ask for help with specific PDFs]
Hi, I'm new to PDFplumber and I'm trying to draw the images detected by the library on the page itself. I do this using the Page.draw_rects() method and use the x0,y0,x1,y1 coordinates obtained from the Page.images properties. But the rects being plotted are mirrored on the x-axis. Am I providing the coordinates in a different order than what is expected by the draw_rects method?
Here is my code:
`rotomecPDF = pdfplumber.open(rotomec)`                               # load pdf
`roto_page6 = rotomecPDF.pages[5]`                                        # select page 6
`pg6_imgs = roto_page6.images`                                              # find images on pg6
`roto_page6.to_image().draw_rects([(i['x0'],i['y0'],i['x1'],i['y1']) for i in pg6_imgs])` # draw a rect for each img 

Here is the pdf 
[Rotomec_5002.pdf](https://github.com/jsvine/pdfplumber/files/13791196/Rotomec_5002.pdf)

And here is the output of the above code. As you can see the images are inverted on the x-axis. 
![Screenshot from 2023-12-29 11-17-27](https://github.com/jsvine/pdfplumber/assets/69640666/17184b10-5cf1-435b-a26f-2e4367ec82ec)
How can I fix this. I know I can write a function to invert the points on the coordinate plane. But I think I'm doing something wrong as the chances of this being a bug in the library are really low.
Thank you in advance and apologies for my English.

**Comment by petermr** [comment]
When I display your post  (in Chrome) and compare with page6 in the PDF (in Acrobat) the images are both in the same orientation.

**Comment by jsvine** [comment]
Hi @yashpundir, a couple of notes to help you on your `pdfplumber` journey:

- `y0` and `y1` are the vertical coordinates in the PDF's *original coordinate space*, where the origin is the *bottom*-left of the page; `pdfplumber`'s methods, however, use the *top*-left of the page as the origin. So you'll want to use the `top` (distance of the top edge of the object from the top of the page) and `bottom` (distance of the bottom edge of the object from the top of the page) attributes instead. I.e., `(x0, top, x1, bottom)`.
- `PageImage`'s drawing methods can accept entire graphical objects, e.g., `im.draw_rect(page.images[0])`

So you could do this:

```python
import pdfplumber
pdf = pdfplumber.open("path/to/file.pdf")
page = pdf.pages[5]
im = page.to_image()
im.draw_rects(page.images)
```

... which returns this:

![image](https://github.com/jsvine/pdfplumber/assets/534702/edf52dbf-9091-4fa8-bf09-d677ffb20315)

**Reply by bikallem** [reply]
Is there a way to extract the said image, i.e. `page.images[0]` into a png/bmp/jpg format?

**Reply by jsvine** [reply]
If your goal is just to extract the images in a PDF, you might have more luck with MuPDF, which includes a specific method for that: https://pymupdf.readthedocs.io/en/latest/document.html#Document.extract_image

`pdfplumber` does not currently have much direct support for image extraction, unfortunately.
